pkg/controller/container-runtime-config/helpers_test.go

```diff
@@ -354,3 +356,90 @@ func TestUpdateRegistriesConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdatePolicyJSON(t *testing.T) {
+	templateConfig := signature.Policy{
+		Default: signature.PolicyRequirements{signature.NewPRInsecureAcceptAnything()},
+		Transports: map[string]signature.PolicyTransportScopes{
+			"docker-daemon": map[string]signature.PolicyRequirements{
+				"": signature.PolicyRequirements{signature.NewPRInsecureAcceptAnything()},
+			},
+		},
+	}
+	buf := bytes.Buffer{}
+	err := json.NewEncoder(&buf).Encode(templateConfig)
+	require.NoError(t, err)
+	templateBytes := buf.Bytes()
+
+	tests := []struct {
+		name             string
+		allowed, blocked []string
+		want             signature.Policy
+	}{
+		{
+			name: "unchanged",
+			want: templateConfig,
+		},
+		{
+			name:    "allowed",
+			allowed: []string{"allow.io"},
+			want: signature.Policy{
+				Default: signature.PolicyRequirements{signature.NewPRReject()},
+				Transports: map[string]signature.PolicyTransportScopes{
+					"atomic": map[string]signature.PolicyRequirements{
+						"allow.io": signature.PolicyRequirements{signature.NewPRInsecureAcceptAnything()},
+					},
+					"docker": map[string]signature.PolicyRequirements{
+						"allow.io": signature.PolicyRequirements{signature.NewPRInsecureAcceptAnything()},
+					},
+					"docker-daemon": map[string]signature.PolicyRequirements{
+						"": signature.PolicyRequirements{signature.NewPRInsecureAcceptAnything()},
+					},
+				},
+			},
+		},
+		{
+			name:    "blocked",
+			blocked: []string{"block.com"},
+			want: signature.Policy{
+				Default: signature.PolicyRequirements{signature.NewPRInsecureAcceptAnything()},
+				Transports: map[string]signature.PolicyTransportScopes{
+					"atomic": map[string]signature.PolicyRequirements{
+						"block.com": signature.PolicyRequirements{signature.NewPRReject()},
+					},
+					"docker": map[string]signature.PolicyRequirements{
+						"block.com": signature.PolicyRequirements{signature.NewPRReject()},
+					},
+					"docker-daemon": map[string]signature.PolicyRequirements{
+						"": signature.PolicyRequirements{signature.NewPRInsecureAcceptAnything()},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := updatePolicyJSON(templateBytes, tt.blocked, tt.allowed)
+			if err != nil {
+				t.Errorf("updatePolicyJSON() error = %v", err)
+				return
+			}
+			gotConf := signature.Policy{}
+			if err := json.Unmarshal(got, &gotConf); err != nil {
+				t.Errorf("error unmarshalling result: %v", err)
+				return
+			}
+			// This assumes a specific order of Registries entries, which does not actually matter; ideally, this would
+			// sort the two arrays before comparing, but right now hard-coding the order works well enough.
+			if !reflect.DeepEqual(gotConf, tt.want) {
+				t.Errorf("updatePolicyJSON() Diff:\n %s", diff.ObjectGoPrintDiff(tt.want, gotConf))
+			}
+			// Ensure that the generated configuration is actually valid.
```

Comment:
This did not do that AFAICS?! `c/image/signature.NewPolicyFromBytes` can be used without the temporary file.

Reply:
Fixed